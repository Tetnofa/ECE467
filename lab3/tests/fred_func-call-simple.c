
int f1(int a1[], int a2) { return 0; }

int f2() {
  int foo[6];
  f1(foo, 3);
}

