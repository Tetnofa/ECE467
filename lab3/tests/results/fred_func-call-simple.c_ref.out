Program [useIO=0](2,0)
	Function[isProto=0]	Int(2,0)
		Identifier[name:f1](2,4)
		Parameter		Int*(2,7)
			Identifier[name:a1](2,11)
		Parameter		Int(2,17)
			Identifier[name:a2](2,21)
		Scope(2,25)
			ReturnStmt(2,27)
				Int Expression(2,34)
					IntConstant[val=0](2,34)
	Function[isProto=0]	Int(4,0)
		Identifier[name:f2](4,4)
		Scope(4,9)
			Array Declaration			Int[6](5,2)
				Identifier[name:foo](5,6)
			Expression Statement(6,2)
				CallExpr(6,2)
					Identifier[name:f1](6,2)
					Argument(6,5)
						Int Expression(6,5)
							Reference(6,5)
								Identifier[name:foo](6,5)
					Argument(6,10)
						Int Expression(6,10)
							IntConstant[val=3](6,10)
